Sources/Extensions/SwiftStdlib/StringExtensions.swift

```diff
@@ -161,9 +161,10 @@ public extension String {
 	///		"john@doe.com".isEmail -> true
 	///
 	public var isEmail: Bool {
-		// http://stackoverflow.com/questions/25471114/how-to-validate-an-e-mail-address-in-swift
-		return range(of: "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}",
-					 options: .regularExpression, range: nil, locale: nil) != nil
+		guard self == trimmingCharacters(in: .whitespacesAndNewlines) else { return false }
+		// http://emailregex.com/
+		let regex = "(?:[\\p{L}0-9!#$%\\&'*+/=?\\^_`{|}~-]+(?:\\.[\\p{L}0-9!#$%\\&'*+/=?\\^_`{|}~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[\\p{L}0-9](?:[a-z0-9-]*[\\p{L}0-9])?\\.)+[\\p{L}0-9](?:[\\p{L}0-9-]*[\\p{L}0-9])?|\\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[\\p{L}0-9-]*[\\p{L}0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)\\])"
```

Comment:
I've read many posts about people saying that email validation is not something that should be done by regex. If you really want to ensure an email is valid you need to send it an email and see that you get a success response from the mail server.
Having such an extensive regex for emails is probably useful but it can potentially be misleading. People might look at this and think that it covers all scenarios and that there will never be false positives or negatives.
My suggestion, therefore, is that we add a note to the comment of the property to make it clear that this should only be used as a _suggestion_ as to whether or not the string is an email.
Personally I'd rather this property be named something like `looksLikeAnEmail` but I think that that would be pretty useless 😝

Reply:
- the purpose of this function is to serve as a local validation step for login and signup screens, to validate the address format before sending a request to server.
- I agree with you, we should add a note to inform user it is just checking the format of the string, and not checking the address itself, how about this one: **Note that this property does not perform authentication to validate the email address. It merely determines whether its format is valid for an email address.**
- About naming, I found [Micosoft](https://docs.microsoft.com/en-us/dotnet/standard/base-types/how-to-verify-that-strings-are-in-valid-email-format) is using the name `IsValidEmail`, maybe we can do so as well 😅

Reply:
- Agreed, I've used this kind of validation before for past projects.
- The note wouldn't be necessarily correct. Even in your new test for `email.@.email.com` you get a false positive, because according to @LucianoPAlmeida 's link of invalid email formats the trailing `.` before the `@` isn't acceptable. That was the point I was trying to make before, which I read in a number of articles, where it's seemingly impossible to have a regex that will tell you for sure if an email address is valid or invalid. So what we're doing here is trying to _verify_, rather than _validate_, the email address. So maybe the note should be something like: **Note that this property does not validate the email address against an email server. It merely attempts to determine whether its format is suitable for an email address.**
- Let's keep the name just to avoid a breaking change 😬

Reply:
I agree too :))
I believe rename is important if it can improve expressivity and explain better the funcionality. Also we don't need to remove `isEmail`, we can just deprecate it and remove in a future version. And add the note on documentation as well.  I like names like you guys suggested `IsValidEmail` or maybe `matchesEmailPattern`. What do you think?

Reply:
The purpose of this property is to determine if some text is an email, locally, before passing it on to the server for further validation. I'm fine with the name `isEmail` or `isValidEmail`. It's an extension on `String` so I believe it's self explanatory that a Swift primitive would not be performing validation server side

Reply:
I see, then I'm deprecating `isEmail` and renaming it to `isValidEmail` to have better consistency with other properties like `isValidURL`

@guykogus I also updated the note 👍